examples/file_server.md

```diff
@@ -2,17 +2,74 @@
 
 ## Concepts
 
+- Use [Deno.open](https://doc.deno.land/builtin/stable#Deno.open) to read a
+  file's content in chunks.
 - Use the Deno standard library
-  [file_server.ts](https://deno.land/std@$STD_VERSION/http/file_server.ts) to
-  run your own file server and access your files from your web browser.
-- Run [Deno install](../tools/script_installer.md) to install the file server
-  locally.
+  [streams module](https://deno.land/std@$STD_VERSION/streams/) to transform a
+  Deno file into a
+  [ReadableStream](https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream).
+- Use Deno's integrated HTTP server to run your own file server.
+
+## Overview
+
+Sending files over the network is a common requirement. As seen in the
+[Fetch Data example](./fetch_data), because files can be of any size, it is
+important to use streams in order to prevent having to load entire files into
+memory.
 
 ## Example
 
-Serve a local directory via HTTP. First install the remote script to your local
-file system. This will install the script to the Deno installation root's bin
-directory, e.g. `/home/alice/.deno/bin/file_server`.
+**Command:** `deno run --allow-read --allow-net file_server.ts`
+
+```ts
+import { readableStreamFromReader } from "https://deno.land/std@$STD_VERSION/streams/mod.ts";
+
+// Start listening on port 8080 of localhost.
+const server = await Deno.listen({ port: 8080 });
+console.log("File server running on http://localhost:8080/");
+
+for await (const conn of server) {
+  const httpConn = Deno.serveHttp(conn);
+  for await (const requestEvent of httpConn) {
```

Comment:
Please see https://deno.land/manual@v1.16.1/runtime/http_server_apis#serving-http

This example will hang the server once a connection is established.

Reply:
Good catch; this bit should be reworked then. @sant123 PRs are welcome!

Reply:
Just created https://github.com/denoland/manual/pull/160